migrations/2024-06-10-084722_create_org_authentication_table/up.sql

```diff
@@ -0,0 +1,18 @@
+-- Your SQL goes here
+CREATE TYPE "AuthMethod" AS ENUM (
+    'okta',
+    'password',
+    'magic_link'
+);
+
+CREATE TABLE IF NOT EXISTS org_authentication_methods (
+    id SERIAL PRIMARY KEY,
+	org_id VARCHAR(64) NOT NULL, 
+    auth_method "AuthMethod" NOT NULL,
+    auth_config JSONB,
+    created_at TIMESTAMP NOT NULL DEFAULT now(),
+    last_modified_at TIMESTAMP NOT NULL DEFAULT now(),
+	CONSTRAINT org_auth_method_unique UNIQUE (org_id, auth_method)
```

Comment:
Where are we going to store info like `totp_required`, `allow_signups`?

If we are going to store them in `auth_config`, why are we using `serde::Value` in the application? I think it should be a proper struct, atleast for the above mentioned fields.

Probably like this:
```rust
struct AuthConfig {
	totp_required: bool,
	allow_signups: bool,
	metadata: Option<serde::Value>
}
```